test/TestProjects/ResourceClients-LowLevel/ResourceClients-LowLevel.json

```diff
@@ -0,0 +1,142 @@
+    "/": {
+      "get": {
+        "description": "Method that belongs to the top level service",
+        "operationId": "Service_GetParameters",
+        "parameters": [],
+        "responses": {
+          "200": {
+            "description": "Success",
+            "schema": {
+              "type":"string"
+            }
+          }
+        }
+      }
+    },
+    "/groups": {
+      "get": {
+        "description": "Get all groups. It is defined in `Group` subclient, but must be promoted to the `Service` client.",
+        "operationId": "Group_ListGroups",
+        "parameters": [],
+        "responses": {
+          "200": {
+            "description": "Success",
+            "schema": {
+              "type":"string"
+            }
+          }
+        }
+      }
+    },
+    "/groups/{groupId}": {
+      "get": {
+        "description": "Get a group. Method should stay in `Group` subclient.",
+        "operationId": "Group_Get",
+        "parameters": [
+          {
+            "$ref": "#/parameters/groupId"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Success",
+            "schema": {
+              "type":"string"
+            }
+          }
+        }
+      }
+    },
+    "/items": {
+      "get": {
+        "description": "Get all groups. It is defined in `Item` subclient, but must be promoted to the `Service` client, because it has neither `groupId` nor `itemId` parameters.",
+        "operationId": "Item_ListAllItems",
+        "parameters": [],
+        "responses": {
+          "200": {
+            "description": "Success",
+            "schema": {
+              "type":"string"
+            }
+          }
+        }
+      }
+    },
+    "/items/{groupId}": {
+      "get": {
+        "description": "Get items in group. It is defined in `Item` subclient, but must be promoted to the `Group` subclient.",
+        "operationId": "Item_ListItems",
+        "parameters": [
+          {
+            "$ref": "#/parameters/groupId"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Success",
+            "schema": {
+              "type":"string"
+            }
+          }
+        }
+      }
+    },
+    "/items/{groupId}/{itemId}": {
+      "get": {
+        "description": "Get an item. Method should stay in `Item` subclient.",
+        "operationId": "Item_Get",
+        "parameters": [
+          {
+            "$ref": "#/parameters/groupId"
+          },
+          {
+            "$ref": "#/parameters/itemId"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Success",
+            "schema": {
+              "type":"string"
+            }
+          }
+        }
+      }
+    }
+  },
+  "definitions": { },
+  "parameters": { 
+    "groupId": {
+      "in": "path",
+      "name": "groupId",
+      "required": true,
+      "type": "string",
+      "x-ms-resource-parameter": true,
```

Comment:
Did we officially close on this name, or is it still something we could change?  I'm wondering if we could call it x-ms-resource-identifier to match what's in the guidelines here? https://azure.github.io/azure-sdk/dotnet_introduction.html#dotnet-subclient-factory-methods-parameters

Parameter could be any parameter to a resource, whereas identifier indicates that it is the unique id of a resource.

Reply:
Also, is there a correct process to follow to add an `x-ms-` autorest extensions?  Would we need to do that for this?

Reply:
We haven't chosen the name and can change it. `x-ms-` attributes can be added and supported per language basis, there are no changes that are required in autorest core.

Reply:
Renamed into `x-ms-resource-identifier`